### ISSUE TITLE
Regenie ignores processing genes, but does not show any error

### ISSUE THREAD

**Issue by soumickmj** [rgcgithub/regenie#507]
Dear regenie Team @rgcgithub ,

A few weeks prior, I discovered a bug while performing burden tests on the DNANexus platform (UK BioBank RAP) using the BGEN files of the whole exome sequences. BGEN files cannot be supplied through the mounted directory "/mnt/project/", but need to be provided as input to the Swiss Army Knife (SAK) - so they are downloaded to the worker.
This resulted in the job consuming excessive memory and consequently failing. This issue has been discussed here: https://github.com/rgcgithub/regenie/issues/365 (I wrote a response there, providing a possible workaround this problem: https://github.com/rgcgithub/regenie/issues/365#issuecomment-1994086435). Another possible solution (I thought at the time) was to use the PLINK files as they worked perfectly, even when I supplied them through the mounted directory. 

Just now, I just discovered a more concerning issue, this time involving the PLINK files. When these files are supplied through the mounted directory "/mnt/project/" to perform burden tests, the operation concludes successfully. However, a critical malfunction occurs: regenie ceases to test genes beyond a certain point. 

For example, during my tests for chromosome 17, the notification "computing association tests...done" was displayed solely for the initial three genes. Consequently, the summary statistics reported an alarmingly reduced number of genes, only including those commencing with letters A to D. Under normal circumstances, one might infer that the absent genes failed to meet the defined filters. Yet, the stark reality is that they were never tested properly. When PLINK files are directly supplied as input files to the SAK and accessed locally within the worker by regenie, "computing association tests...done" message is shown for all the genes. 

This discrepancy poses a significant challenge, primarily because it manifests no explicit error, thereby misleading users into believing the job has been executed successfully. In essence, this is not the case.

I must add that the challenges outlined are not isolated to regenie executed through the Swiss Army Knife, but similar problems also occur when utilising Jupyter Notebook or interactive sessions to run regenie. 

I am unsure whether this issue pertains exclusively to regenie in conjunction with DNANexus, or whether it concerns regenie's interaction with a stream of data over a network (e.g., network mounts). My personal belief is that it is the latter. If this is indeed the case, it might pose a significant issue for the regenie user base, especially since it fails to notify users of any error, creating the impression that all operations were executed successfully.

I would like to request that this issue be addressed with the utmost urgency.

Thanks.

**Comment by joellembatchou** [comment]
Hi,

So when using streaming, the requirement is that the file is only examined once in sequential order.  So this was fine when you ran it with single variants in PLINK BED format as you only analyze each variant once. However, with gene-based tests, you can have the same variants with annotations across multiple genes (as specified  in the setlist file) and the order of variants listed in that file may not be sequential. Hence, streaming will not work in such cases. 
So in summary, to use streaming, you'd need to check in the setlist file that (1) there are no variants that are assigned across multiple genes; (2) the ordering of the variants across genes matches with that in the BIM file.

As far as your run on chromosome 17 with a reduced number of genes analyzed, can you include the full log file?
 
Cheers,
Joelle

**Comment by akhilpampana** [comment]
Has this issue been resolved? I am observing a similar trend where I am getting some of the genes removed even though they have variants that cross MAC > 5 filtering criteria at least for some of the masks.

**Comment by joellembatchou** [comment]
@akhilpampana  Please submit a new issue with details of what you are experiencing (eg log file). 
Thanks,
Joelle
